lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +214,44 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self._seed_urls = config.get_seed_urls()
+        self._config = config
+        self.urls = []
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        href = article_bs.get('href')
+        if isinstance(href, str) and \
+                href.startswith('https://chelny-izvest.ru/news/') and href.count('/') == 5:
+            return href
+        return ''
 
     def find_articles(self) -> None:
         """
         Finds articles
         """
-        pass
+        with concurrent.futures.ThreadPoolExecutor(max_workers=10) as executor:
```

Comment:
I used multithreading because I don't know how else to shorten the runtime :( If I shorten the sleep time, the site blocks requests. At the lowest possible sleep time, CI thinks it's still a long running function.

Reply:
how long does it take to collect 10 articles on your local computer without multithreading?

Reply:
with time.sleep(random.uniform(0.5, 1)) and timeout=5:
to collect 10 articles: 17 sec 969 ms
to collect 100 articles: 2 min 58 sec